adapter/config/default_config.go

```diff
@@ -180,7 +180,14 @@ var defaultConfig = &Config{
 					Name:                 "APIM Publisher",
 					Issuer:               "https://localhost:9443/publisher",
 					ValidateSubscription: true,
-					CertificateAlias:     "",
+					CertificateAlias:     "publisher_certificate_alias",
+					CertificateFilePath:  "/home/wso2/security/truststore/wso2carbon.pem",
+				},
+				{
+					Name:                 "APIM APIkey",
+					Issuer:               "",
+					ValidateSubscription: true,
```

Comment:
The only use of this block is to enable or disable validation subscription for apim apikeys, right? I feel other fields in this config will not make much sense to a user
When cpenabled, are there any actual scenarios where users need to skip subscription validation? Does APIM too support turning off subscription validation for API keys?

Reply:
The other scenarios would be to set a different certificate for API key. In that case, that cert would get stored under the alias defined in this block.

We enable subscription validation by default when CP is enabled. Did not come across a option to skip sub validation in APIM side. The only similar option was to disable security.

Reply:
Does APIM support setting a separate cert for APIKeys? Isn't it issued by the same cert used for JWTs in APIM?

Reply:
With the default configs, the Resident KM only sends a jwks endpoint. We retrive the cert when the initial cert arrives, and store the cert using the kid of the token as the alias. Unless we change the default configs, access tokens may have that can be updated time to time is what I got to know during a team discussion. And found this https://is.docs.wso2.com/en/latest/references/concepts/authentication/jwks/